client.go

```diff
@@ -595,6 +612,159 @@ func (c *Client) FollowersOff(channel string) {
 	c.Say(channel, "/followersoff")
 }
 
+func parseVipsOrModsMsg(content string) []string {
+	index := strings.IndexRune(content, ':')
+	if index == -1 {
+		// https://play.golang.org/p/2LNyI4GbQuh
+		return []string{}
+	}
+	content = strings.Trim(content[index+1:], " ")
+	if content[len(content)-1] == '.' {
+		content = content[:len(content)-1]
+	}
+	return strings.Split(content, ", ")
+}
+
+// GetVips run twitch command `/vips` with the given channel in argument and returns a string slice of all vips
+func (c *Client) GetVips(channel string, timeout time.Duration) ([]string, error) {
+	// make sure the channel is always lowercase to prevent duplicate entries in the map.
+	channel = strings.ToLower(channel)
+	// channel for message callback
+	ch := make(chan NoticeMessage, 1)
+	// if we should make a new callback group
+	var callback *noticeCallback
+	var msg NoticeMessage
+	var msgSet bool
+
+	defer func() {
+		// close the channel
+		close(ch)
+		// concurrent array writes
+		callback.mutex.Lock()
+		defer callback.mutex.Unlock()
+		// simple filter to remove the channel from the array of channels, needed cause with timeout one function might end before another due to the timeout and then we do not want to delete the entry from the map.
+		for i, v := range callback.chans {
+			if v == ch {
+				callback.chans = append(callback.chans[:i], callback.chans[i+1:]...)
+			}
+		}
+		// if we are the last one clean up the parent map.
+		if len(callback.chans) == 0 {
+			c.vipsMtx.Lock()
+			delete(c.vipsChan, channel)
+			c.vipsMtx.Unlock()
+		}
+	}()
+
+	c.vipsMtx.RLock()
+	if v, ok := c.vipsChan[channel]; ok {
+		c.vipsMtx.RUnlock()
+		v.mutex.Lock()
+		// if the callback has already proceeded
+		if v.resp != nil {
+			v.mutex.Unlock()
+			msg = *v.resp
+			msgSet = true
+		}
+		v.mutex.Unlock()
+		callback = v
+	} else {
+		c.vipsMtx.RUnlock()
+		c.vipsMtx.Lock()
+		callback = &noticeCallback{
+			chans: []chan NoticeMessage{ch},
+			mutex: &sync.Mutex{},
+		}
+		c.vipsChan[channel] = callback
+		c.vipsMtx.Unlock()
+		c.Say(channel, "/vips")
+	}
+
+	if !msgSet {
+		if timeout == 0 {
+			msg = <-ch
+		} else {
+			select {
+			case <-time.After(timeout):
+				return nil, ErrRequestTimedout
+			case msg = <-ch:
+			}
+		}
+	}
+
+	return parseVipsOrModsMsg(msg.Message), nil
+}
+
+// GetMods run twitch command `/mods` with the given channel in argument and returns a string slice of all mods
+func (c *Client) GetMods(channel string, timeout time.Duration) ([]string, error) {
```

Comment:
This should be an option on the client instead that you can configure, this would be too confusing for an end-user to decide which timeout is approriate.